src/components/App.js

```diff
@@ -1,14 +1,154 @@
 import React from 'react';
 import { hot } from 'react-hot-loader/root';
-import macaroon from '../assets/img/Macaroonicon.png';
-
-function App() {
-  return (
-    <div className="app">
-      <h1 className="title">Shmotify.</h1>
-      <img src={macaroon} alt="" />
-    </div>
-  );
+import { checkURIforError, getTokenFromURI } from '../config/authConfig';
+import { Dashboard } from './Dashboard';
+import { UserPlaylists } from './UserPlaylists';
+import { Sidebar } from './Sidebar';
+import { Welcome } from './Welcome';
+import { LoginLink } from './LoginLink';
+
+
+class App extends React.Component {
+  state = {
+    token: '',
+    user: {
+      name: '',
+      error: false,
+      errorMessage: '',
+    },
+    playlists: {
+      items: [],
+      nextPlaylistsEndpoint: null,
+      error: false,
+      errorMessage: '',
+    },
+  }
+
+  componentDidMount() {
+    const authError = checkURIforError();
+    const token = getTokenFromURI();
+    this.setAuthError(authError);
+    if (token) {
+      this.setToken(token);
+      this.getProfile(token);
+      this.getPlaylists(token);
+    }
+  }
+
+  componentDidUpdate() {

```

Comment:
This is my way of dealing with the fact that Spotify's API returns large arrays in small slices, with each wrapped in a pager object that gives you the URL to the next slice. On update, grab the next slice from the URL it just returned until it doesn't return another URL. This is also why `nextPlaylistEndpoint` is initialized to `null`, since eventually it is going to return `null` anyway.